expr/plan.go

```diff
@@ -124,11 +157,11 @@ func consumeArg(args []*expr, j int, exp argType) (int, error) {
 // it also makes sure the kwarg has not been consumed already via the kwargs map
 // (it would be an error to provide an argument twice via the same keyword,
 // or once positionally and once via keyword)
-func consumeKwarg(optArgs []optArg, namedArgs map[string]*expr, k string, seenKwargs map[string]struct{}) error {
+func consumeKwarg(namedArgs map[string]*expr, k string, optArgs []arg, seenKwargs map[string]struct{}) error {
```

Comment:
I like your suggestions around refactoring these consume funcs. but two notes:

1) how would consumeKwarg know about the set of optional args that is allowed? we still have to pass that in I think.

2) why export the consume functions? seems to me all parsing/planning will happen from within this package